genie-web/src/main/java/com/netflix/genie/web/rpc/grpc/services/impl/v4/GRpcJobTerminationServiceImpl.java

```diff
@@ -0,0 +1,107 @@
+/*
+ *
+ *  Copyright 2018 Netflix, Inc.
+ *
+ *     Licensed under the Apache License, Version 2.0 (the "License");
+ *     you may not use this file except in compliance with the License.
+ *     You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *     Unless required by applicable law or agreed to in writing, software
+ *     distributed under the License is distributed on an "AS IS" BASIS,
+ *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *     See the License for the specific language governing permissions and
+ *     limitations under the License.
+ *
+ */
+package com.netflix.genie.web.rpc.grpc.services.impl.v4;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.netflix.genie.proto.JobKillListenerRequest;
+import com.netflix.genie.proto.JobKillListenerResponse;
+import com.netflix.genie.proto.JobTerminationServiceGrpc;
+import com.netflix.genie.web.rpc.grpc.interceptors.SimpleLoggingInterceptor;
+import com.netflix.genie.web.services.JobKillCallbackService;
+import io.grpc.stub.StreamObserver;
+import lombok.extern.slf4j.Slf4j;
+import net.devh.springboot.autoconfigure.grpc.server.GrpcService;
+
+/**
+ * Extension of {@link JobTerminationServiceGrpc.JobTerminationServiceImplBase}
+ * to provide functionality for killing jobs running on agents.
+ *
+ * @author standon
+ * @since 4.0.0
+ */
+@GrpcService(
+    value = JobTerminationServiceGrpc.class,
+    interceptors = {
+        SimpleLoggingInterceptor.class,
+    }
+)
+@Slf4j
+public class GRpcJobTerminationServiceImpl extends JobTerminationServiceGrpc.JobTerminationServiceImplBase {
+
+    private final JobKillCallbackService jobKillCallbackService;
+
+    /**
+     * Constructor.
+     *
+     * @param jobKillCallbackService The implementation of the {@link JobKillCallbackService} to use
+     */
+    public GRpcJobTerminationServiceImpl(final JobKillCallbackService jobKillCallbackService) {
+        this.jobKillCallbackService = jobKillCallbackService;
+    }
+
+    /**
+     * Register to get a callback when a job is killed.
+     *
+     * @param request          Request to listen for job registerCallback
+     * @param responseObserver The response observer
+     */
+    @Override
+    public void registerForKillNotification(
+        final JobKillListenerRequest request,
+        final StreamObserver<JobKillListenerResponse> responseObserver
+    ) {
+        jobKillCallbackService.registerCallbackForKill(
+            request.getJobId(),
+            new JobKillCallback(responseObserver, request.getJobId())
+        );
+    }
+
+    //Testing utility
+    @VisibleForTesting
+    JobKillCallback getCallBack(
```

Comment:
This seems like a useless method. Why don't you just make the class itself package protected and invoke the constructor directly fo testing?

Reply:
Tried that approach earlier but for some reason was not able to do that in groovy. So, created this. Moved back to package level class and have created a junit based test. Works now.